src/c_api/c_api.cc

```diff
@@ -464,7 +464,10 @@ XGB_DLL int XGDMatrixCreateFromDense(char const *data,
   float missing = GetMissing(config);
   auto n_threads = OptionalArg<Integer, int64_t>(config, "nthread", 0);
   xgboost_CHECK_C_ARG_PTR(out);
-  *out = new std::shared_ptr<DMatrix>(DMatrix::Create(&adapter, missing, n_threads));
+  auto data_split_mode =
```

Comment:
This line should be above the check on `out`.

Reply:
Fixed!